registry/api/errcode/errors.go

```diff
@@ -69,20 +69,28 @@ func (ec *ErrorCode) UnmarshalText(text []byte) error {
 // WithDetail creates a new Error struct based on the passed-in info and
 // set the Detail property appropriately
 func (ec ErrorCode) WithDetail(detail interface{}) Error {
```

Comment:
@duglin In this case I think it would make the most sense to rename the methods. The ErrorCode type really serves as a factory for Error types.

It's not that big a deal and @stevvooe is OK with it so I'm not going to force the issue. It just jumped out at me when I was reading the code to make use of it in notary.